src/react/header/Header.tsx

```diff
@@ -0,0 +1,145 @@
+import clsx from 'clsx'
+import type { ReactNode } from 'react'
+import { useEffect, useRef } from 'react'
+import { MODlogo } from './MODlogo'
+
+interface NavigationOptions {
+  key?: string
+  content?: ReactNode | string
+  href?: string
+  active?: boolean
+  attributes?: Omit<JSX.IntrinsicElements['a'], 'href'>
+}
+
+interface HeaderContentProps {
+  serviceName?: string
+  serviceUrl?: string
+  menuButtonLabel?: string
+  menuButtonText?: string
+  navigation?: NavigationOptions[]
+  navigationClasses?: string
+  navigationLabel?: string
+}
+
+interface HeaderProps extends HeaderContentProps {
+  homepageUrl?: string
+  assetsPath?: string
+  productName?: string
+  containerClasses?: string
+  classes?: string
+  extraProps?: JSX.IntrinsicElements['header']
```

Comment:
Does it make sense to extend from this instead since they are for the root element? I think the current way doesn't quite work with e.g. `data-testid`.

Reply:
Also `JSX.IntrinsicElements['header']` includes `className` and `role` which overwrite the values we set. I think we need to exclude `className` at least as it breaks stuff (changing the `role` may be a non-issue since it does what you'd probably expect it to...)